src/data/mv_polynomial/counit.lean

```diff
@@ -0,0 +1,64 @@
+/-
+Copyright (c) 2020 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+
+import data.mv_polynomial.basic
+
+/-!
+## Counit morphisms for multivariate polynomials
+
+One may consider the ring of multivariate polynomials `mv_polynomial A R` with coefficients in `R`
+and variables indexed by `A`. If `A` is not just a type, but an algebra over `A`,
+then there is a natural surjective algebra homomorphism `mv_polynomial A R →ₐ[R] A`
+obtained by `X a ↦ a`.
+
+### Main declarations
+
+* `mv_polynomial.acounit R A` is the natural surjective algebra homomorphism `mv_polynomial A R →ₐ[R] A`
+     obtained by `X a ↦ a`
+* `mv_polynomial.counit` is an “absolute” variant with `R = ℤ`
+* `mv_polynomial.counit_nat` is an “absolute” variant with `R = ℕ`
+
+-/
+
+namespace mv_polynomial
+open function
+
+variables (A B R : Type*) [comm_semiring A] [comm_semiring B] [comm_ring R] [algebra A B]
+
+/-- `mv_polynomial.acounit R A` is the natural surjective algebra homomorphism
+`mv_polynomial A R →ₐ[R] A` obtained by `X a ↦ a`.
+
+See `mv_polynomial.counit` for the “absolute” variant with `R = ℤ`,
+and `mv_polynomial.counit_nat` for the “absolute” variant with `R = ℕ`. -/
+noncomputable def acounit : mv_polynomial B A →ₐ[A] B :=
```

Comment:
It's a bit annoying that the variable names don't match the docstring.

Reply:
Should be fixed now.